src/device.jl

```diff
@@ -103,6 +103,14 @@ function filter_free(devlist)
         end
         ccall(("nvmlShutdown", libnvml), UInt32, ())
         return freelist
+    elseif isempty(nvidia_smi)
+        Base.warn_once(
+        """
+        Neither NVML nor nvidia-smi are available.
+        Please check your setup and install either one.
+        `CUDArt.filter_free` will not work.
```

Comment:
Ah you mean reorder the if-clause. Yeah it's probably clearer to have the predicates in the other order but that won't change the semantics. Will change that tomorrow.